### DISCUSSION TITLE
É possível validar qualquer Nota Fiscal?

### DISCUSSION THREAD

**Discussion by juaryjunior** [TadaSoftware/PyNFe#269] [General]
Olá amigos, sou novo nesse mundo por isso já peço desculpas se minha questão for obvia.
Eu tenho uma missão que é poder validar notas fiscais, só preciso saber se é Falsa ou Verdadeira, mas de forma massiva por isso que encontrei essa biblioteca do Python e achei muito interessante, queria muito saber se através dessa biblioteca eu consigo fazer esse trabalho.
Obs: a ideia é validar qualquer nota fiscal, mesmo se não foi emitida para meu CPF ou CNPJ.

Parabéns pelo brilhante trabalho.

**Comment by juniortada** [comment]
Boa noite @juaryjunior o que você considera como nota fiscal falsa ou verdadeira?
O que você vai validar? Chave da NF-e? xml? Status da nota?
Você pode validar se é uma chave válida, se o xml é válido, se a assinatura é válida, etc.
Mas só o que garante que é uma nota enviada para a SEFAZ e tem algum status atual(autorizado/cancelado/etc)
é fazer a consulta dela no webservice adequado.
Acho que o importante seria consultar a nota https://github.com/TadaSoftware/PyNFe/wiki/Consulta-Nota

**Reply by juaryjunior** [reply]
boa noite @juniortada, primeiramente obrigado pelo retorno.
Talvez não me expressei bem na minha questão, trabalho em uma área que faz reembolso, e para isso eu preciso checar se a nota fiscal que meu cliente enviou não é uma "nota fria", talvez esse link que me passou me ajude de alguma forma, pois no geral seria apenas consultar se a nota existe na receita.
